processing-service-core/api/models/db/repositoryConfigStore.js

```diff
@@ -0,0 +1,186 @@
+const repositoryConfigTable = databaseDefinition.define(repositoryConfigTableDetails.name,
+    tableDefinition,
+    { tableName:  repositoryConfigTableDetails.tableName, timestamps: false });
+
+/**
+ * Deletes a repository config matching the specified parameters.
+ * @param repositoryId {String} the repository ID identifying the config that should be deleted.
+ * @param tenanatId {String} the tenant ID identifying the config that should be deleted.
+ * @param key {String} the key identifying the config that should be deleted.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on result of delete.
+ *  Resolved promise will return an empty object.
+ *  Rejected promise will pass an ApiError with either type set to ITEM_NOT_FOUND if no repository config with the specified repository ID
+ *  and key could be found or type DATABASE_UNKNOWN_ERROR for any other kind of failure.
+ */
+function deleteRepositoryConfig(repositoryId, tenantId, key) {
+    var deferredDelete = Q.defer();
+
+    repositoryConfigTable.destroy({ where: { repositoryId: repositoryId, key: key, tenanatId: tenantId }})
+        .then(function(deleteResult){
+            if(deleteResult < 1) {
+                logger.debug("No repository config found for repositoryId '"+repositoryId+"', key '"+ key +"', and tenantId '"+ tenantId +"' found.");
+                deferredDelete.reject(apiErrorFactory.createNotFoundError(
+                    "Repository config specified for delete using repositoryId '"+repositoryId+"', key '"+key+"', and tenantId '"+ tenantId +"' was not found."));
+                return;
+            }
+            deferredDelete.resolve({});
+        })
+        .catch(function(errorResponse){
+            logger.error("Failure occurred during delete of repository config with repositoryId '"
+                +repositoryId+"', key '"+ key +"', and tenantId '"+ tenantId +"': "+errorResponse.toString());
+            deferredDelete.reject(apiErrorFactory.createDatabaseUnknownError("Failure during delete of repository config with" +
+                " repositoryId '" +repositoryId+ "', key '" +key+"', and tenantId '"+ tenantId +"'."));
+        });
+
+    return deferredDelete.promise;
+}
+
+/**
+ * Retrieves the value for a repository config matching the specified parameters.
+ * @param repositoryId {String} the repository ID identifying the tenant config.
+ * @param tenanatId {String} the tenant ID identifying the config that should be deleted.
+ * @param key {String} the key identifying the repository config.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on the result of retrieval.
+ *  Resolved promise will return the value on the matched tenant config.
+ *  Rejected promise can pass an ApiError with type set to either DATABASE_UNKNOWN_ERROR or ITEM_NOT_FOUND if unable to find
+ *  the specified tenant config.
```

Comment:
"if unable to find the specified tenant config": I'm not sure if that is correct or not